spec/models/client_spec.rb

```diff
@@ -101,9 +105,10 @@
   describe "Client prefixes transfer" do
     let!(:prefixes) { create_list(:prefix, 3) }
     let!(:prefix) { prefixes.first }
-    let!(:client_prefix)  { create(:client_prefix, client: client, prefix: prefix) }
-    let!(:provider_prefix)  { create(:provider_prefix, provider: provider, prefix: prefix) }
+     ### Order is important in creating prefixes relations
```

Comment:
This comment is a little confusing, it's not so much the order is important, but the fact the client_prefix needs the provider_prefix id in order to relate it, if you left it in the original order while trying to reference provider_prefix.uid this would fail to build.

Reply:
actually it wouldn't fail, that is why i did it caught it the first time. The factory has a 'create' strategy so if there is no `provider_prefix_id` it will create one, but not the one that one wants